src/PAModelTests/FilePathTests.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.PowerPlatform.Formulas.Tools;
+using Microsoft.PowerPlatform.Formulas.Tools.Utility;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.IO;
+
+namespace PAModelTests
+{
+    [TestClass]
+    public class FilePathTests
+    {
+        // If the length of the escaped file name is beyond 60, then we truncate the length to limit it to 60.
+        [DataTestMethod]
+        [DataRow("Long*Name*File*Path*Validation*Tests***.fx.yaml", FilePath.MaxFileNameLength)]
```

Comment:
This should also test the hash, we should compare against an expected actual name.